sdk/ai/azure-ai-inference/src/samples/java/com/azure/ai/inference/usage/TraceStreamingChatCompletionsToolCallSample.java

```diff
@@ -0,0 +1,314 @@
+import com.azure.core.util.IterableStream;
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class TraceStreamingChatCompletionsToolCallSample {
+    private static final String APP_NAMESPACE = "contoso-flight-info-app";
+
+    /**
+     * @param args Unused. Arguments to the program.
+     */
+    @SuppressWarnings("try")
+    public static void main(final String[] args) {
+        final OpenTelemetrySdk telemetry = configureOpenTelemetry();
+        final ChatCompletionsClient client = createChatCompletionClient();
+        final Tracer tracer = telemetry.getTracer(APP_NAMESPACE);
+
+        final Span span = tracer.spanBuilder("Contoso.App").startSpan();
+        try (AutoCloseable scope = span.makeCurrent()) {
+            final List<ChatRequestMessage> messages = new ArrayList<>();
+            messages.add(new ChatRequestSystemMessage("You an assistant that helps users find flight information."));
+            messages.add(new ChatRequestUserMessage("What is the next flights from Seattle to Miami?"));
+            final GetFlightInfoFunction function = new GetFlightInfoFunction();
+
+            final IterableStream<StreamingChatCompletionsUpdate> toolCallChunks = client.completeStream(new ChatCompletionsOptions(messages).setTools(function.toolDefinitions()));
+            final ChunksMerged toolCallChunksMerged = ChunksMerged.create(toolCallChunks);
+            final ChatCompletionsToolCall toolCall = toolCallChunksMerged.asTooCall();
+            messages.add(toAssistantMessage(toolCall));
+
+            final ChatRequestToolMessage toolMessage = function.invoke(toolCall);
+            messages.add(toolMessage);
+
+            final IterableStream<StreamingChatCompletionsUpdate> modelResponseChunks = client.completeStream(new ChatCompletionsOptions(messages).setTools(function.toolDefinitions()));
+            final ChunksMerged modelResponseChunksMerged = ChunksMerged.create(modelResponseChunks);
+            System.out.println("Model response: " + modelResponseChunksMerged.content);
+            span.end();
+        } catch (Exception e) {
+            span.setAttribute("error.type", e.getClass().getName());
+            span.end();
+            throw new RuntimeException(e);
+        } finally {
+            telemetry.close();
```

Comment:
it's actually unnecessary - it uses JVM shutdown hook to finish and export everything when the JVM is shutting down.

Reply:
I see, okay, I did that because IntelliJ warned about not closing this `AutoClosable.`

Reply:
`close` is removed in https://github.com/anuchandy/azure-sdk-for-java/pull/18